packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2640,10 +2641,13 @@ class BuildOwner {
           }
           return true;
         }());
-        if (!kReleaseMode && debugProfileBuildsEnabled) {
+        final bool isTimelineTracked = !kReleaseMode && _isProfileBuildsEnabledFor(element.widget);
```

Comment:
What's the cost of this?

Reply:
All the costs are listed in the description and performance measurements are in the comments.  The TLDR is that with it for the Flutter Gallery executing this versus not sometimes measures faster.  So the added overhead is below the variance.

Reply:
Please check this patch on the internal mobile harness testing on a low end device. Flutter Gallery even on a Moto G4 is not representative, particularly for things that will do extra work on every frame, compared to the internal low end Android device for customer: money.

Reply:
responded below.